app/scripts/modules/core/src/managed/managed.dataSource.ts

```diff
@@ -42,5 +50,24 @@ module(MANAGED_RESOURCES_DATA_SOURCE, []).run([
       afterLoad: addManagedMetadataToResources,
       defaultData: { applicationPaused: false, hasManagedResources: false, resources: [] },
     });
+
+    ApplicationDataSourceRegistry.registerDataSource({
+      key: 'environments',
+      sref: '.environments',
+      category: DELIVERY_KEY,
+      optional: true,
+      optIn: true,
+      hidden: true,
+      icon: 'fa fa-fw fa-xs fa-code-branch',
+      loader: loadEnvironments,
+      onLoad: addEnvironments,
+      defaultData: {
+        applicationPaused: false,
+        hasManagedResources: false,
+        environments: [],
+        artifacts: [],
+        resources: [],
```

Comment:
is duplicating the `resources` data into this data source a temporary/convenience thing, or do you intend on it being permanent?

Reply:
Well originally temporary/convenience, but thinking about it some more I think it might be more permanent.

I don't like the duplication either but because the `managedResource` data source is `visible: false`, it doesn't contribute to the application ready state, so I don't have a guarantee that it will be available when my route loads.